syntax = "proto3";

package SFG.Proto;

message MsgContent {
  string username = 1;
  string msgText = 2;
}

message SendMessageRequest {
  MsgContent msg = 1;
}
message SendMessageResponse {
  bool success = 1;
  string errorMsg = 2;
}

message GetAllMessagesRequest {
}
// This shall be returned by req-rep with above request or by pub-sub
message AllMessages {
  repeated MsgContent msgList = 1;
}

// This shall be only used for errors before the server doesn't respond anymore
message ReqRepErrorMessage {
}
